File: 백준/Bronze/15161. Grass Cutting/Grass Cutting.py
import sys
input=lambda:sys.stdin.readline().rstrip()
k=int(input())
arr=[[1]*10 for _ in range(10)]
for i in range(k):
    lines=[*map(int,input().split())]
    rows,cols=lines[:3],lines[3:]
    for j in range(10):
        for k in range(10):
            if j+1 in rows:
                arr[j][k]=1
            elif k+1 in cols:
                arr[j][k]=1
            else:
                arr[j][k]+=1
for i in range(10):
    print(*arr[i])

File: 백준/Bronze/15161. Grass Cutting/Grass Cutting.cc
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int k;
    cin >> k;
    vector<int> rows(3), cols(3);
    vector<vector<int>> arr(10, vector<int>(10, 1));
    for (int i = 0; i < k; i++) {
        for (int j = 0; j < 3; j++) {
            cin >> rows[j];
        }
        for (int j = 0; j < 3; j++) {
            cin >> cols[j];
        }
        for (int j = 0; j < 10; j++) {
            for (int k = 0; k < 10; k++) {
                if (find(rows.begin(), rows.end(), j + 1) != rows.end()) {
                    arr[j][k] = 1;
                } else if (find(cols.begin(), cols.end(), k + 1) != cols.end()) {
                    arr[j][k] = 1;
                } else {
                    arr[j][k] += 1;
                }
            }
        }
    }
    for (int i = 0; i < 10; i++) {
        for (int j = 0; j < 10; j++) {
            cout << arr[i][j] << ' ';
        }
        cout << "\n";
    }
    return 0;
}